db/background_mgr_attachment_migration.go

```diff
@@ -0,0 +1,351 @@
+		}
+	}
+	defer persistClusterStatus()
+
+	var processFailure error
+	failProcess := func(err error, format string, args ...interface{}) bool {
+		processFailure = err
+		terminator.Close()
+		base.WarnfCtx(ctx, format, args...)
+		return false
+	}
+
+	callback := func(event sgbucket.FeedEvent) bool {
+		docID := string(event.Key)
+		collection := db.CollectionByID[event.CollectionID]
+		base.TracefCtx(ctx, base.KeyAll, "[%s] Received DCP event %d for doc %v", migrationLoggingID, event.Opcode, base.UD(docID))
+
+		// Ignore documents without xattrs, to avoid processing unnecessary documents
+		if event.DataType&base.MemcachedDataTypeXattr == 0 {
+			return true
+		}
+
+		// Don't want to process raw binary docs
+		// The binary check should suffice but for additional safety also check for empty bodies
+		if event.DataType == base.MemcachedDataTypeRaw || len(event.Value) == 0 {
+			return true
+		}
+
+		// We only want to process full docs. Not any sync docs.
+		if strings.HasPrefix(docID, base.SyncDocPrefix) {
+			return true
+		}
+
+		a.DocsProcessed.Add(1)
+		syncData, _, _, err := UnmarshalDocumentSyncDataFromFeed(event.Value, event.DataType, collection.userXattrKey(), false)
+		if err != nil {
+			failProcess(err, "error unmarshaling document %s: %v", base.UD(docID), err)
+		}
+
+		if syncData == nil || syncData.Attachments == nil {
+			// no attachments to migrate
+			return true
+		}
+
+		collCtx := collection.AddCollectionContext(ctx)
+		collWithUser := &DatabaseCollectionWithUser{
+			DatabaseCollection: collection,
+		}
+		// xattr migration to take place
+		err = collWithUser.MigrateAttachmentMetadata(collCtx, docID, event.Cas, syncData)
+		if err != nil {
+			failProcess(err, "error migrating document attachment metadata for doc: %s: %v", base.UD(docID), err)
+		} else {
+			a.DocsChanged.Add(1)
+		}
+		return true
+	}
+
+	bucket, err := base.AsGocbV2Bucket(db.Bucket)
+	if err != nil {
+		return err
+	}
+
+	currCollectionIDs := db.GetCollectionIDs()
+	checkpointPrefix := db.MetadataKeys.DCPCheckpointPrefix(db.Options.GroupID) + "att_migration:"
+
+	// check for mismatch in collection id's between current collections on the db and prev run
+	err = a.resetDCPMetadataIfNeeded(ctx, db.MetadataStore, db, checkpointPrefix, currCollectionIDs)
+	if err != nil {
+		return err
+	}
+
+	a.SetCollectionIDs(currCollectionIDs)
+	dcpOptions := getMigrationDCPClientOptions(currCollectionIDs, db.Options.GroupID, checkpointPrefix)
+	dcpFeedKey := GenerateAttachmentMigrationDCPStreamName(a.MigrationID)
+	dcpClient, err := base.NewDCPClient(ctx, dcpFeedKey, callback, *dcpOptions, bucket)
+	if err != nil {
+		base.WarnfCtx(ctx, "[%s] Failed to create attachment migration DCP client: %v", migrationLoggingID, err)
+		return err
+	}
+	base.InfofCtx(ctx, base.KeyAll, "[%s] Starting DCP feed %q for attachment migration", migrationLoggingID, dcpFeedKey)
```

Comment:
I think should Debug at best, since we already marked that we are starting the process in `Init`, unless that resume doesn't pass through the code.